### PR TITLE
pg_partman as optional dependency

### PR DESCRIPTION
**Pull request by ChuckHend**
pg_partman is only required when using partitioned queues, which is not the default queue in the extension. Let's not make it a hard requirement, which will make it easier to install.

Returns to SQL caller: 
```text
SQL Error [22000]: ERROR: missing dependency: pg_partman
```

And in the postgres logs:

```text
2023-08-07 15:25:33.333 CDT [5263] WARNING:  pg_partman not installed. Install https://github.com/pgpartman/pg_partman and then run `CREATE EXTENSION pg_partman;`
2023-08-07 15:25:33.334 CDT [5263] ERROR:  missing dependency: pg_partman
2023-08-07 15:25:33.334 CDT [5263] STATEMENT:  select pgmq_create_partitioned('z')
```

Also clearing up some clippy warnings in the PR

### CODE REVIEW

**Comment by samay-sharma** [comment]
I was thinking of this today too. What's the behavior now? Does it error out when trying to create a partitioned queue?

**Comment by ChuckHend** [comment]
> I was thinking of this today too. What's the behavior now? Does it error out when trying to create a partitioned queue?

Yes it will error. I was thinking of making return a more informative error message, and adding some doc around it.

**Review by nhudson — APPROVED** [review]

**Review by sjmiller609 — APPROVED** [review]
clean

File: pgmq.control [hunk 1]
```diff
@@ -1,6 +1,5 @@
-comment = 'Distributed message queues'
+comment = 'A lightweight message queue. Like AWS SQS and RSMQ but on Postgres.'
 default_version = '@CARGO_VERSION@'
 module_pathname = '$libdir/pgmq'
 relocatable = false
 superuser = false
-requires = 'pg_partman'
```
Comment (ianstanton) [inline pgmq.control:6]: :+1: I'm in favor of dropping this requirement

**Review by ianstanton — APPROVED** [review]

**Pull request merged**